aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -126,6 +126,12 @@ class TORCH_API Dispatcher final {
   template<class Return, class... Args>
   Return callWithDispatchKey(const TypedOperatorHandle<Return (Args...)>& op, DispatchKey dispatchKey, Args... args) const;
 
+  // Like callWithDispatchKey, but provides the dispatch key as a set of keys.
+  // The dispatch key with highest priority is used.
+  // See Note [Plumbing Keys Through The Dispatcher]
+  template<class Return, class... Args>
+  Return callWithDispatchKey(const TypedOperatorHandle<Return (Args...)>& op, DispatchKeySet dispatchKeySet, Args... args) const;
```

Comment:
This new API probably could land as its own PR.

I'm not sure the inlining will work out. The optimizer would need to be smart enough to know that constructing a singleton bit set and then immediately extracting the first bit is a no op. That seems like a bit of a lift... though optimizers have worked out harder stuff certainly!

Reply:
Hmmm I don't think I'm following.

This new API is actually required in order to avoid doing extra work. Previously, `call()` would generate the current DispatchKey, and pass it along to `callWithDispatchKey()`. This has the downside of throwing away the rest of the keys in the set.

Now that we want to avoid having to re-calculate the full DispatchKeySet, I updated `call()` to grab compute the current DispatchKeySet without bothering to compute the highest priority key, and pass that entire set through `callWithDispatchKey()` -> `KernelFunction.call()`.

I should probably rename the new version of `callWithDispatchKey` to `callWithDispatchKeySet` though.

Reply:
We are in violent agreement. It looks like I misinterpreted your earlier comment 

> callWithDispatchKey, redispatch and callBoxed now each have two variants, one taking in a DispatchKey and one taking in a DispatchKeySet (callBoxed's first variant doesn't take in anything). They are functionally similar, and I need the second variant, but I kept the first around to avoid being BC-breaking. Still, that gives us two pretty much repeated functions. I could write the first variant to call the second, I'd just need to benchmark it to ensure it gets inlined properly.

as referring to this code, but I guess it refers to somewhere else